testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestAvroSchemaEvolutionWithPartitioning.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertQueryFailure;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryExecutor.query;
+import static io.trino.tests.product.TestGroups.AVRO;
+import static io.trino.tests.product.utils.QueryExecutors.onHive;
+import static java.lang.String.format;
+
+public class TestAvroSchemaEvolutionWithPartitioning
+        extends BaseTestAvroSchemaEvolution
+{
+    public TestAvroSchemaEvolutionWithPartitioning()
+    {
+        super("product_tests_avro_table_with_partitioning");
+    }
+
+    @Override
+    protected void createAndLoadTestTable(String tableName)
+    {
+        query(format(
+                "CREATE TABLE %s (" +
+                        "  dummy_col VARCHAR," +
+                        "  partition_col VARCHAR " +
+                        ")" +
+                        "WITH (" +
+                        "  format='AVRO', " +
+                        "  partitioned_by=ARRAY['partition_col'], " +
+                        "  avro_schema_url='%s'" +
+                        ")",
+                tableName,
+                ORIGINAL_SCHEMA));
+        query(format("INSERT INTO %s VALUES ('stringA0', 0, 'stringB0')", tableName));
+    }
+
+    @Test(groups = AVRO)
```

Comment:
Can we move some of the similar tests to `BaseTestAvroSchemaEvolution`. Like we could have same set of columns for both the tests and one would be marked as partition in TestAvroSchemaEvolutionWithPartitioning. WDYT ?

Reply:
Yes I thought of that. And made an approach. But effect provided very unclear test code.
Like we need to provide list of columns, mark one of them as partition and also provide example data for inserts.
Then all of assertions would refer to this list, sometimes one example row in some cases two.
Let's discuss it offline.

Reply:
Ok as discussed offline will try to make only partition columns configurable.

Reply:
Applied looks ok, let me know your opinion.